controllers/flinkcluster_util.go

```diff
@@ -42,7 +42,7 @@ const (
 	ControlRetries            = "retries"
 	ControlMaxRetries         = "3"
 
-	SavepointTimeoutSec = 60
+	SavepointTimeoutSec = 900 // 15 mins
```

Comment:
Can we make it configurable as a field in the job spec?

Reply:
This might not be the best case, I increased it for the moment but I think we need to check the jobmanager's API to see SP status in the next SP PR. Do you think its bad we increased it to 15 mins? Is there a case where some1 will want to take SP every <15mins?

Reply:
This constant is actually the "minimal interval for triggering 2 savepoints", right? The name "Timeout" could be confusing, it might be mis-interpreted as "the timeout for taking a savepoint (before considering it as a failure)".

It is hard to determine the value. For example, I just took a savepoint 10 mins ago, but now I want to update my job, and I don't want to lose the state for the recent 10 mins, so I want it to take another savepoint before the update. Why do we need to introduce an arbitrary limit here?

Reply:
There are currently three time variables related to the savepoint.
* `SavepointAgeForJobUpdateSec`: savepoint age limit required for update progress
* `SavepointRequestRetryIntervalSec`: Retry interval in case of failure to create savepoint for update
* `SavepointTimeoutSec`: savepoint creation timeout

Occasionally, the savepoint creation could not proceed anymore due to an error, but the job manager returned its status normally. In that case, SavepointTimeoutSec is being used to handle the timeout.  To handle jobs that require a long time to create savepoints, it would be better to change this variable to be user-configurable and set its default value large enough.

https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/f3e73cadc44850d7abf6a6b6a612b2807afa779e/controllers/flinkcluster_util.go#L45-L51

Reply:
I found that it is possible to set the checkpoint timeout with the Flink configuration. In my opinion, it would be better to remove the Flink operator's savepoint timeout handling routine to handle the second issue and guide the timeout of the Flink configuration.

note: https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/config.html#execution-checkpointing-timeout

Reply:
I think this might be better solved in another PR, this one provides a mitigation for clusters that SP takes more than a few seconds.
I think we should discuss how we want to solve this in another issue and consider to get the info about the SP (is there an active one, was it timeout, etc..) from the jobmanager itself.
I can make this one a part of the crd for now and then later delete it when it will no longer be needed (in another PR)
WTYT @elanv @functicons

Reply:
SGTM, let's address it in another PR.

Reply:
Sorry for the late response. When a checkpoint timeout occurs in Flink jobmanager, the savepoint state falls to "falied", so I don't think the first savepoint needs to be identified. The second issue is occurring because the default Flink checkpoint timeout is 10 minutes, but SavepointTimeoutSec is less than that. I think it's okay to handle that part in another PR.

Reply:
`SavepointTimeoutSec` is just savepoint timeout and `autoSavepointSeconds` is the savepoint trigger interval as  you mentioned.